test/0.8.9/utils/pausable-util.test.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "forge-std/Test.sol";
+import { PausableUntil } from "contracts/0.8.9/utils/PausableUntil.sol";
+
+contract ExposedPausableUntil is PausableUntil {
+    function setPausedState(uint256 _resumeSince) public {
+        _setPausedState(_resumeSince);
+    }
+
+    function pauseUntil(uint256 _pauseUntilInclusive) public {
+        _pauseUntil(_pauseUntilInclusive);
+    }
+
+    function pauseFor(uint256 _duration) public {
+        _pauseFor(_duration);
+    }
+
+    function resume() public {
+        _resume();
+    }
+
+    function checkResumed() public view {
+        _checkResumed();
+    }
+
+    function checkPaused() public view {
+        _checkPaused();
+    }
+}
+
+contract PausableUntilTest is Test {
```

Comment:
can we have a fuzzed test that `pauseFor` effectively is dual to `pauseUntil`:

I mean the following in pseudocode
```solidity
pausableUntil1.pauseFor(x)
pausableUntil2.pauseUntill(block.timestamp + x - 1)

assertEq(
   pausableUntil1.getResumeSinceTimestamp(), 
   pausableUntil2.getResumeSinceTimestamp()
)
```